src/Microsoft.ML.Transforms/PermutationFeatureImportance.cs

```diff
@@ -0,0 +1,281 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Internallearn;
+using Microsoft.ML.Runtime.Internal.Utilities;
+
+namespace Microsoft.ML.Transforms
+{
+    internal static class PermutationFeatureImportance<TResult>
+    {
+        public static ImmutableArray<TResult>
+            GetImportanceMetricsMatrix(
```

Comment:
If you have time, can you refactor this method into at least two chunks, initialization and  for loop on top of permutation routine. Or maybe just put permutation routine into separate function.
In current state it's quite hard to read. (~200 lines)